### PR TITLE
Add eslint

### LINKED ISSUE #1085: Add eslint pre-commit hook
- [x] Create a configuration for eslint. (0d) - @shreyamalviya 
- [x] Resolve all existing eslint errors. (0d) - @shreyamalviya 
- [x] Add a pre-commit hook to run eslint on commit. (0d) - @shreyamalviya 
- [x] Fail travis build if any eslint errors are introduced. (0d) - @shreyamalviya 
- [x] Update developer documentation. (0d) - @shreyamalviya

### PR DESCRIPTION
**Pull request by shreyamalviya**
Fixes #1085

### CODE REVIEW

File: docs/content/development/setup-development-environment.md [hunk 1]
```diff
@@ -23,3 +23,11 @@ This means setting up an environment with Linux 32/64-bit with Python installed
...
 
 To setup a working front environment, run the instructions listed in the [`readme.txt`](https://github.com/guardicore/monkey/blob/master/monkey/monkey_island/readme.txt)
+
+## Pre-commit
+
+Pre-commit is a multi-language package manager for pre-commit hooks. It will run a set of checks when you attempt to commit. If your commit does not pass all checks, it will be reformatted and/or you'll be given a list of errors and warnings that need to be fixed before you can commit.
+
+Our CI system runs the same checks when pull requests are submitted. This system may report that the build has failed if the pre-commit hooks have not been run or all issues have not been resolved.
+
+To install and configure pre-commit, run `pip install --user pre-commit`. Next, go to the top level directory of this repository and run `pre-commit install`. Pre-commit will now run automatically whenever you `git commit`.
```
Comment (VakarisZ) [inline docs/content/development/setup-development-environment.md:33]: Wasn't this change part of pre-commit PR?

**Review by VakarisZ — APPROVED** [review]

File: docs/content/development/setup-development-environment.md (hunk 1 above)
Comment (mssalvatore) [reply docs/content/development/setup-development-environment.md:33]: Yeah, we put it in `deployment_scrips/README.md`. It makes sense that it should go here too.

Comment (mssalvatore) [reply docs/content/development/setup-development-environment.md:33]: Perhaps we should shorten the one in `deployment_scrips/README.md` to reference this one to remove duplication.

**Review by mssalvatore — CHANGES_REQUESTED** [review]
Where is the eslint config? Did I just miss it?

File: docs/content/development/setup-development-environment.md (hunk 1 above)
Comment (shreyamalviya) [reply docs/content/development/setup-development-environment.md:33]: Opened #1099

**Comment by shreyamalviya** [comment]
@mssalvatore `.eslintrc` is in `monkey_island/cc/ui`

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**